lampod/src/liquidity.rs

```diff
@@ -1,35 +1,271 @@
-use std::sync::Arc;
+use std::{sync::Arc, time::Duration};
 
-use lampo_common::keys::LampoKeysManager;
-use lightning_liquidity::{events::Event, LiquidityManager};
+use lampo_common::{
+    bitcoin::hashes::{sha256, Hash},
+    conf::LampoConf,
+    error,
+    keys::LampoKeysManager,
+    ldk::{
+        invoice::{Bolt11Invoice, InvoiceBuilder, RouteHint, RouteHintHop, RoutingFees},
+        ln::{channelmanager::MIN_FINAL_CLTV_EXPIRY_DELTA, msgs::SocketAddress},
+        sign::KeysManager,
+    },
+    secp256k1::{PublicKey, Secp256k1},
+};
+use lightning_liquidity::{
+    events::Event,
+    lsps0::ser::RequestId,
+    lsps2::{event::LSPS2ClientEvent, msgs::OpeningFeeParams},
+    LiquidityManager,
+};
 
-use crate::{chain::LampoChainManager, ln::LampoChannel};
+use crate::{
+    chain::LampoChainManager,
+    ln::{LampoChannel, LampoChannelManager},
+    LampoDaemon,
+};
 
-type LampoLiquidity =
+pub type LampoLiquidity =
     LiquidityManager<Arc<LampoKeysManager>, Arc<LampoChannel>, Arc<LampoChainManager>>;
 
-struct LampoLiquiditySource {
-    lampo_liquidity_manager: Arc<LampoLiquidity>,
+// Configured when we are acting as a client sourcing lsp from
+// a different node, here provider is someone who is providing us liquidity.
+pub struct LSPProvider {
```

Comment:
I do not need to read the comment to see that the `LSPProvider` is `LSPClient`, the `P` in the `LSP` is already mean `Provider`

Reply:
Nice catch, thanks. Something like `LampoLiquidityProvider` would work?

Reply:
The problem is the suffix `Provider`, iirc we already discuss this to call `LSPManager`